File: C#/0606. Construct String from Binary Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public string Tree2str(TreeNode t) {
        if(t==null){return "";}
        string rep=t.val.ToString();
        if(t.left==null && t.right==null){return rep;}
        else if (t.left!=null && t.right==null){return rep+"("+Tree2str(t.left)+")";}
        else if (t.left==null && t.right!=null){return rep+"()"+"("+Tree2str(t.right)+")";}
        else{return rep+"("+Tree2str(t.left)+")"+"("+Tree2str(t.right)+")";}    
    }
}

File: 20190519python/0606. Construct String from Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def tree2str(self, t: TreeNode) -> str:
        if not t:
            return ""
        
        rep=str(t.val)
        if t.left and t.right:
            rep+='('+self.tree2str(t.left)+')'
            rep+='('+self.tree2str(t.right)+')'
        elif not t.left and t.right:
            rep+='()'
            rep+='('+self.tree2str(t.right)+')'
        elif t.left and not t.right:
            rep+='('+self.tree2str(t.left)+')'
            
        return rep
